File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
#include<bits/stdc++.h>

class Solution {
public:
    string longestCommonPrefix(vector<string>& s) {
        
        string ans="";
        
        sort(s.begin(),s.end());
        
        string f=s[0];
        string l=s[s.size()-1];
        
        int i=0;
        while(i<min(f.size(),l.size())){
            if(f[i]!=l[i]){
                return ans;
            }
            ans+=f[i];
            i++;
        }
        
        return ans;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        
        Arrays.sort(strs);
        
        String first = strs[0];
        String last = strs[strs.length-1];
        
        String res="";
        
        int i=0;
        while(i < Math.min(first.length(),last.length())){
            if(first.charAt(i) == last.charAt(i)){
                res += first.charAt(i);
            }else{
                break;
            }
            i++;
        }
        
        return res;
    }
}